sequencer/src/bin/espresso-bridge.rs

```diff
@@ -195,10 +220,16 @@ async fn deposit(opt: Deposit) -> anyhow::Result<()> {
                 continue;
             }
         };
-        let Some(l1_finalized) = header.l1_finalized() else {
-            continue;
+        let l1_finalized = if opt.require_l1_finality {
+            if let Some(l1_finalized) = header.l1_finalized() {
+                l1_finalized.number
+            } else {
+                continue;
+            }
+        } else {
+            header.l1_head()
         };
-        if l1_finalized.number >= l1_block {
+        if l1_finalized >= l1_block {
```

Comment:
Does this actually do anything? It makes the bridge tool exit sooner. But those funds still won't be available on Espresso until Espresso has seen the deposit finalized on L1. So we aren't able to start building blocks any earlier. If anything this might increase the chance we start the builder too soon and start building invalid blocks (which will stall consensus) since the builder isn't funded

Reply:
Right, I didn't consider that, will rollback this part.